guillotina/api/login.py

```diff
@@ -84,3 +87,253 @@ class Refresh(Service):
         await notify(UserRefreshToken(user, jwt_token))
 
         return {"exp": data["exp"], "token": jwt_token}
+
+
+@configure.service(
+    context=IContainer,
+    method="POST",
+    permission="guillotina.Logout",
+    name="@logout",
+    summary="Logout application",
+    allow_access=True,
+)
+@configure.service(
+    context=IApplication,
+    method="POST",
+    permission="guillotina.Logout",
+    name="@logout",
+    summary="Logout application",
+    allow_access=True,
+)
+class Logout(Service):
+    async def __call__(self):
+        user = get_authenticated_user()
+        try:
+            await user.logout(self.request)
+        except AttributeError:
+            pass
+
+@configure.service(
+    context=IContainer,
+    name="@users/{user}/reset-password",
+    method="POST",
+    permission="guillotina.Public",
+    responses={
+        "200": {
+            "description": "Reset password",
+        }
+    },
+    summary="Reset password",
+    allow_access=True,
+)
+class ResetPasswordUsers(Service):
+    async def __call__(self):
+        user_id: str = self.request.matchdict["user"]
+        actual_user = get_authenticated_user()
+        if actual_user.id == user_id:
+            # Self setting password
+            # Payload : {
+            #   'old_password': 'secret',
+            #   'new_password': 'verysecret',
+            # }
+
+            data = await self.request.json()
+            try:
+                await actual_user.set_password(
+                    data.get('new_password', None),
+                    old_password=data.get('old_password', None)
+                )
+            except AttributeError:
+                raise HTTPNotAcceptable()
+        else:
+            # We validate with recaptcha
+            validator = RecaptchaValidator()
```

Comment:
I think this should be a utility that can be customized and maybe the default implementation could be a `return True`

Reply:
I feel that this should be a parameter on the service view to protect the view with a CSRF like aproach. This is a quick solution that should be evolvoved to a more generic way. I think its out of the scope of this PR to provide more options that Recaptcha.

Reply:
At least would be great to be customizable. I'm ok with Recaptcha as a default implementation as long as I can replace it if I want :P

Reply:
Right now you can disable it if you don't have the recaptcha param on settings. Looking forward for a PR to define Recaptcha/CSRF/generalized validation on view settings!

Reply:
Oh, I see, if it's not defined returns true